src/components/atoms/SearchBar/index.jsx

```diff
@@ -0,0 +1,131 @@
+//#region > Imports
+//> React
+// Contains all the functionality necessary to define React components
+import React from "react";
+// React PropTypes
+import PropTypes from "prop-types";
+//> MDB
+// "Material Design for Bootstrap" is a great UI design framework
+import {
+  MDBNavbar,
+  MDBNavbarBrand,
+  MDBNavbarNav,
+  MDBNavItem,
+  MDBNavLink,
+  MDBNavbarToggler,
+  MDBCollapse,
+  MDBContainer,
+  MDBDropdown,
+  MDBDropdownItem,
+  MDBDropdownToggle,
+  MDBDropdownMenu,
+  MDBSmoothScroll,
+  MDBIcon,
+  MDBSelect,
+  MDBSelectInput,
+  MDBSelectOption,
+  MDBSelectOptions,
+  MDBBtn,
+} from "mdbreact";
+//#endregion
+
+//#region > Components
+/**
+ * @class A component which contains a search bar
+ */
+class SearchBar extends React.Component {
+  state = {
+    filter: "",
+    usernames: [],
+  };
+
+  handleSelection = (event, value) => {
+    if (event === "user") {
+      window.open("/u/" + value, "_self");
+    } else if (event === "search_page") {
+      window.open("search?q=", "_self");
+    }
+  };
+
+  search = (event) => {
+    let value = document
+      .getElementById("selectSearchInput")
+      .getAttribute("value");
+
+    // 'keypress' event misbehaves on mobile so we track 'Enter' key via 'keydown' event
+    if (event.key === "Enter") {
+      event.preventDefault();
+      event.stopPropagation();
+      this.handleSelection("search_page", value);
+    } else {
+      this.setState({
+        filter: value,
+      });
+    }
+  };
+
+  getUsernameList = () => {
+    const { globalFunctions } = this.props;
+
+    globalFunctions.users().then((usernames) => {
+      this.setState({
+        usernames,
+      });
+    });
+  };
+
+  render() {
+    const fuzzysort = require("fuzzysort");
```

Comment:
Cant this be moved to the imports?

Reply:
> Can this be moved to the imports?

@Aichnerc

Reply:
Should be possible. I will move it so that we have all imports in the import region.